openpype/hosts/maya/plugins/publish/convert_gltf_shader.py

```diff
@@ -0,0 +1,78 @@
+import os
+from maya import cmds
+import pyblish.api
+
+from openpype.pipeline import publish
+
+
+class ConvertGLSLShader(publish.Extractor):
+    """
+    Converting StingrayPBS material to GLSL Shaders
+    specially for the glb export through Maya2GLTF plugin
+
+    """
+    order = pyblish.api.ExtractorOrder - 0.1
+    hosts = ["maya"]
+    label = "Convert StingrayPBS to GLTF"
+    families = ["gltf"]
+    optional = True
+
+    def process(self, instance):
+        meshes = cmds.ls(instance, type="mesh", long=True)
+        self.log.info("meshes: {}".format(meshes))
+        # load the glsl shader plugin
+        cmds.loadPlugin("glslShader.mll", quiet=True)
+
+        for mesh in meshes:
+
+            # create glsl shader
+            glsl = cmds.createNode('GLSLShader')
+            glsl_shadingGrp = cmds.sets(name=glsl + "SG", empty=True,
+                                        renderable=True, noSurfaceShader=True)
+            cmds.connectAttr(glsl + ".outColor",
+                             glsl_shadingGrp + ".surfaceShader")
+
+            # load the maya2gltf shader
+            maya_dir = os.getenv("MAYA_APP_DIR")
+            ogsfx = maya_dir + "/maya2glTF/PBR/shaders/glTF_PBR.ogsfx"
```

Comment:
This is incorrect - this would _only_ work if the user had a local install of `maya2glTF`. This should either be configurable in settings or in some other way be defined.

We tended to check the path relative to the maya2gltf module.

Reply:
i am not sure about this actually, As I go through maya2glTF documentation and installation, it seems that it is recommended to be put into the default maya_app_dir plug-in. of course it can be configurable. and we can allow this configurable in the settings. This is the simplest way to define the environment for it but if the user doesn't define anything in the enivornment, I think I will just leave it as os.getenv("MAYA_APP_DIR") by default